create privacy policy page with attractive and responsive UI (Closes #237)

Harshit457 opened the pull request:
# **Fixes Issue🛠️**  

Closes #237  

# **Description👨‍💻**  

- Created a new **Privacy Policy** page that matches the existing **UI theme**.  
- Ensured **proper layout**, using **light colors and purple-blue text** for consistency.  
- Added **a section for an image** on the left side to make the page visually appealing.  
- Created a new **route `/privacy`** for easy access to the Privacy Policy.  
- Added a **Privacy Policy button** in the **footer** for quick navigation.  
- Made the page **responsive** to ensure it looks great on different screen sizes.

# **Type of Change📄**  

- [x] New feature (non-breaking change which adds functionality)  
- [x] Style (non-breaking change which improves website style or formatting)  
- [x] Documentation (non-breaking change which updates or adds documentation)  
- [x] Responsiveness (non-breaking change which improves UI/UX on different screen sizes)  

# **Checklist✅**  

- [x] I am an Open Source contributor  
- [x] I have performed a self-review of my code  
- [x] My code follows the style guidelines of this project  
- [x] I have commented on my code, particularly in hard-to-understand areas  
- [x] I have installed packages from `frontend` folder by `npm ci` instead of `npm i`  

# **Screenshots/GIF📷**  
📌 **Screenshot 1: **
![privacy policy 1 ](https://github.com/user-attachments/assets/240b9fbb-3649-404f-be85-5dee671a9cd1)

📌 **Screenshot 2: Footer with Privacy Policy Button**
![footer with privacy policy](https://github.com/user-attachments/assets/43391e85-6737-4e9b-ab67-7eb316cd05ec)

PratikMane0112 commented on frontend/package-lock.json:
```diff
@@ -23343,6 +23343,20 @@
         "is-typedarray": "^1.0.0"
       }
     },
+    "node_modules/typescript": {
```
Unnecessarily added

PratikMane0112 reviewed (commented):
- Plz avoid doing `npm i` instead to `npm ci` while commit, it gets added unnecessary stuff to `package-lock.json`
- You can also make a checkbox to accept `terms & conditions`
- Rest all UI is good

Harshit457 commented:
> * Plz avoid doing `npm i` instead to `npm ci` while commit, it gets added unnecessary stuff to `package-lock.json`
> * You can also make a checkbox to accept or `terms & conditions`
> * Rest all UI is good



> * Plz avoid doing `npm i` instead do `npm ci` while commit, it gets added unnecessary stuff to `package-lock.json`
> * You can also make a checkbox to accept or `terms & conditions`
> * Rest all UI is good

Sure I will do the changes please add level label on the issue and PR

PratikMane0112 commented:
> Sure I will do the changes please add level label on the issue and PR

Sure, level will added after merge

PratikMane0112 commented:
You can also add below points -
- Google OAuth Authorization
- Also some extra points for doctor & patients data (explore website or Readme)
- No need to make the checkbox (ignore my [prev comment](https://github.com/PratikMane0112/TelMedSphere/pull/238#issuecomment-2633864599))

Harshit457 commented:
![Screenshot 2025-02-04 204536](https://github.com/user-attachments/assets/6c60fb4a-9e3c-4db8-a892-139a9bdc1795)
I have added details of Patient and Doctor by taking reference from readme and website and also added the google OAuth Authorization part and this time i didnt do any changes in package.lock.json

PratikMane0112 commented on frontend/package.json:
```diff
@@ -72,4 +72,4 @@
     "tailwindcss": "^3.4.17"
   }
-}
+}
```
Changed

PratikMane0112 commented on frontend/package-lock.json:
```diff
@@ -25572,4 +25572,4 @@
       }
     }
   }
-}
```
Changed

Harshit457 commented:
@PratikMane0112 please check this now  refer to the latest commit
